python/thunder/rdds/series.py

```diff
@@ -636,6 +636,111 @@ def query(self, inds, var='inds', order='F', onebased=True):
+            generated by a SeriesBlockingStrategy to be close to the requested size.
+            A tuple of positive ints will be interpreted as "splits per dimension". Only certain patterns of splits
+            are valid to convert Series back to Blocks; see docstring above. These splits will be passed into a
+            SeriesBlockingStrategy that will be used to generate the returned blocks.
+            If an instance of SeriesBlockingStrategy is passed, it will be used to generate the returned Blocks.
+
+        Returns
+        -------
+        Blocks instance
+        """
+        from thunder.rdds.imgblocks.strategy import BlockingStrategy, SeriesBlockingStrategy
+        if isinstance(blockSizeSpec, SeriesBlockingStrategy):
+            blockingStrategy = blockSizeSpec
+        elif isinstance(blockSizeSpec, basestring) or isinstance(blockSizeSpec, int):
+            blockingStrategy = SeriesBlockingStrategy.generateFromBlockSize(self, blockSizeSpec)
+        else:
+            # assume it is a tuple of positive int specifying splits
+            blockingStrategy = SeriesBlockingStrategy(blockSizeSpec)
+
+        blockingStrategy.setSource(self)
+        avgSize = blockingStrategy.calcAverageBlockSize()
+        if avgSize >= BlockingStrategy.DEFAULT_MAX_BLOCK_SIZE:
+            # TODO: use logging module here rather than print
+            print "Thunder WARNING: average block size of %g bytes exceeds suggested max size of %g bytes" % \
+                  (avgSize, BlockingStrategy.DEFAULT_MAX_BLOCK_SIZE)
+
+        returnType = blockingStrategy.getBlocksClass()
+        blockedRdd = self.rdd.map(blockingStrategy.blockingFunction)
+        # since our blocks are likely pretty big, try setting 1 partition per block
+        groupedRdd = blockedRdd.groupByKey(numPartitions=blockingStrategy.nblocks)
+        # <key>, <val> at this point is:
+        # <block number>, <[(series key, series val), (series key, series val), ...]>
+        simpleBlocksRdd = groupedRdd.map(blockingStrategy.combiningFunction)
+        return returnType(simpleBlocksRdd, dims=self.dims, nimages=len(self.index), dtype=self.dtype)
+
+    def saveAsBinarySeries(self, outputdirname, overwrite=False):
+        """Writes out Series-formatted data.
+
+        This method (Series.saveAsBinarySeries) writes out binary series files using the current partitioning
+        of this Series object. (That is, if mySeries.rdd.getNumPartitions() == 5, then 5 files will be written
+        out, one per partition.) The records will not be resorted; the file names for each partition will be
+        taken from the key of the first Series record in that partition. If the Series object is already
+        sorted and no records have been removed by filtering, then the resulting output should be equivalent
+        to what one would get from calling myImages.saveAsBinarySeries().
```

Comment:
Sounds good, will add a "see also" or similar. Agreed, calling `convertImagesToSeries` should be at least somewhat faster.